Sprint-2/extend/format-time.js

```diff
@@ -22,3 +23,26 @@ console.assert(
   currentOutput2 === targetOutput2,
   `current output: ${currentOutput2}, target output: ${targetOutput2}`
 );
+ 
+// My code starts here
+
+const currentOutput3 = formatAs12HourClock("14:00");
```

Comment:
Great! What would happen if the user sent "28:17"?

Reply:
Ha ha I did not write code to throw an error message for the edge case where users input a time that is not possible. A 24hr clock can only go up to to 23:59 so if they input hours beyond 23 or minutes beyond 59, there should be a message to them with feedback and instructions.

Out of curiosity, should I write code that actually prevents the function from running or is an error message sufficient? I'm thinking about instances where that input data might be stored in a database and I should disallow that.

Reply:
Think back to the forms project in the first module. We write validations on the inputs and reject values that don't match the pattern. So long as there is a user, it's better to communicate with them in this way: throw an error and pass a message explaining what to do. 

Sometimes there isn't a user - there's just messy data you have to reconcile. In this case you might choose to clean and transform the data yourself.